### PR TITLE
Add HTTPS port in generated containers by Kubernetes

### LINKED ISSUE #29999: Kubernetes Extension Should be Aware of Standard TLS Properties
### Description

When generating Kubernetes resources using the [Kubernetes extension](https://quarkus.io/guides/deploying-to-kubernetes)
I would expect that when set an the tls port, for example:  `quarkus.http.ssl-port=443`

This will be reflected in the generated `Deployment` resource, thus an additional port will be added to the container definition:

```
    - containerPort: 443
      name: tls
      protocol: TCP
```

In addition to that, if I turn off insecure requests, using:
`quarkus.http.insecure-requests=disabled`
The http port will **not** be generated anymore:

```
 - containerPort: 80
   name: http
   protocol: TCP
```

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
These changes will add an additional container port HTTPS in the generated manifests:

```yaml
containers:
        - image: ...
          imagePullPolicy: IfNotPresent
          name: kubernetes-kind
          ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            - containerPort: 8443
              name: https
              protocol: TCP
```

By default, the Ingress and Route resources will use the "http" port. However, as part of these changes, I've added a new property to select between "https" or "http", or any other that user might have added as part of the configuration. Example:

```
quarkus.kubernetes.ingress.target-port=https
quarkus.openshift.route.target-port=https
```

Finally, note that the https container won't be added for the Knative resources since there seems to be a limitation at Knative side where only one port can be used.

Fix https://github.com/quarkusio/quarkus/issues/29999

### CODE REVIEW

**Comment by geoand** [comment]
I'll review this one closely tomorrow

**Comment by geoand** [comment]
The test failure does seem related

**Comment by Sgitario** [comment]
> The test failure does seem related

Yep, it seems to be caused by a bug in Dekorate that I have just reported: https://github.com/dekorateio/dekorate/issues/1123. What puzzles me is that the issue was random. Anyway, I already have a fix for Quarkus, so I will update this PR soon.

**Comment by Sgitario** [comment]
PR updated

**Comment by Sgitario** [comment]
Spotted this new issue in Dekorate related to the Route port: https://github.com/dekorateio/dekorate/issues/1124. 
PR updated by adding another workaround.

**Comment by geoand** [comment]
Should we perhaps wait to get a Dekorate release?

**Comment by Sgitario** [comment]
> Should we perhaps wait to get a Dekorate release?

Taking into account that this is very usual, I would say better to work around the issues and when bumping Dekorate with the fixes, remove the workarounds. Otherwise, it would delay many features in Quarkus.

**Comment by geoand** [comment]
> Otherwise, it would delay many features in Quarkus

What do you have in mind?

**Comment by Sgitario** [comment]
> > Otherwise, it would delay many features in Quarkus
> 
> What do you have in mind?

No, I meant other changes in these Quarkus extensions. That this is what I used to do when I found blockers.

**Comment by geoand** [comment]
I would prefer we fix the issues in Dekorate first as this is a new feature so there is no real rush to get it in, but I won't object, as this is definitely useful work.

**Comment by Sgitario** [comment]
> I would prefer we fix the issues in Dekorate first as this is a new feature so there is no real rush to get it in, but I won't object, as this is definitely useful work.

I will be working on the issues in Dekorate. If we address them quickly, I'm also favored to wait for the next Dekorate release including these fixes and remove the workarounds from this pull request.

**Comment by geoand** [comment]
Thanks!

**Review by iocanel — COMMENTED** [review]
I hope that once we get a dekorat release out, we'll have less things to review.
So, I think that we should wait. After all the relevant PR are already merged on dekorate.

**Comment by Sgitario** [comment]
Pull request that bumps Dekorate: https://github.com/quarkusio/quarkus/pull/30351

**Comment by Sgitario** [comment]
PR updated using the latest Dekorate release with the mentioned issues fixed.

**Review by geoand — APPROVED** [review]

**Comment by Sgitario** [comment]
@iocanel any change you can take a look into this, please?

**Comment by iocanel** [comment]
Sure

**Review by iocanel — CHANGES_REQUESTED** [review]
Looks good to me.

I only have a minor naming concern on whether the property should be called `target-port` or `target-port-name`. In dekorate we went with the former, however I am having second thoughts, as in quarkus there are areas we are going to have one or the other (e.g. #30547).

**Comment by Sgitario** [comment]
> Looks good to me.
> 
> I only have a minor naming concern on whether the property should be called `target-port` or `target-port-name`. In dekorate we went with the former, however I am having second thoughts, as in quarkus there are areas we are going to have one or the other (e.g. #30547).

I'm a bit reluctant of adding two properties for the port number and the port name. What do you think of keeping the `target-port` property, but instead of using a String, using a "IntOrString"-like type? Similar to how Fabric8 Kubernetes Client does.

**Comment by Sgitario** [comment]
ping @iocanel

**Comment by iocanel** [comment]
> * there is already an existing property for routes with the name "target-port"
touche

Most importantly, it seems that in the kubernetes context `target-port` most of the times refers to name while `port` refers to 
number. Something, that I think you also mentioned in one of your comments  (but can't find it now).

**Review by iocanel — APPROVED** [review]
lgtm

**Pull request merged**
